src/SamplesServiceModelGenerator.Tests/Swagger/ParserTests.cs

```diff
@@ -0,0 +1,132 @@
+        private static readonly char[] ItemSeparators = { ';' };
+        private static readonly Regex EnumRegex = new Regex(@"^\s*(?<enumName>[^= ]+)\s*=\s*(?<fieldName>[^. ]+)\s*\.\s*(?<valueList>[^ ]+)\s*$", RegexOptions.Compiled);
+        private static readonly char[] ListSeparators = { ',', ' ' };
+
+        private Parser testparser = new Parser();
+        private Api parseOutput = new Api();
+        private JObject json = new JObject();
+
+        private string LoadStringFromUrl(string url)
+        {
+            var uri = new Uri(url);
+
+            Console.WriteLine($"Fetching {uri} ...");
+
+            using (var client = new HttpClient())
+            {
+                var response = client.GetAsync(uri).Result;
+                return response.Content.ReadAsStringAsync().Result;
+            }
+        }
+
+        [SetUp]
+        public void ForEachTest()
+        {
+            jsonText = LoadStringFromUrl(baseUrl);
+
+            testparser = new Parser {
+                EnumOverrides = _enums
+                    .Split(ItemSeparators, StringSplitOptions.RemoveEmptyEntries)
+                    .Select(s => EnumRegex.Match(s))
+                    .Where(m => m.Success)
+                    .ToDictionary(
+                        m => $"{m.Groups["fieldName"].Value.Trim()}.{string.Join(",", m.Groups["valueList"].Value.Split(ListSeparators, StringSplitOptions.RemoveEmptyEntries))}",
+                        m => new SamplesServiceModelGenerator.Swagger.Enum(
+                            new Property { Name = m.Groups["enumName"].Value.Trim() },
+                            new Property { Name = m.Groups["enumName"].Value.Trim() },
+                            m.Groups["valueList"].Value.Split(ListSeparators, StringSplitOptions.RemoveEmptyEntries)))
+            };
+
+            try
+            {
+                json = JObject.Parse(jsonText);
+            }
+            catch (Exception ex)
+            {
+                throw new Exception("Test Swagger is not a Valid Json", ex);
+            }
+
+            parseOutput = testparser.Parse(jsonText, baseUrl);
+        }
+
+        [Test]
+        public void ParseBaseUrl()
+        {
+            Assert.AreEqual(baseUrl, parseOutput.BaseUrl);
+        }
+
+        [Test]
+        public void ParseTitle()
+        {
+            Assert.AreEqual(json["info"]["title"].ToString(), parseOutput.Title);
+        }
+
+        [Test]
+        public void ParseDefinitions()
+        {
+            parseOutput.Definitions.Should().NotBeEmpty();
+
+            parseOutput.Definitions.Count().Equals(json["definitions"].Count());
+
+            JObject expectedDefinitions = (JObject)json["definitions"];
+            List<String> expectedProperties = ["message", "localizationKey", "localizationParameters", "requestId", "id", "name", "description", "canEditAllData", "samplingLocationGroups", "auditAttributes"];
```

Comment:
As you've already done in other places, you can use implicit and built in types here

```suggestion
            var expectedDefinitions = (JObject)json["definitions"];
            List<string> expectedProperties = ["message", "localizationKey", "localizationParameters", "requestId", "id", "name", "description", "canEditAllData", "samplingLocationGroups", "auditAttributes"];
```